libs/ledger-live-common/src/families/icon/js-transaction.ts

```diff
@@ -0,0 +1,45 @@
+import { BigNumber } from "bignumber.js";
+import type { IconAccount, Transaction } from "./types";
+import { defaultUpdateTransaction } from "@ledgerhq/coin-framework/bridge/jsHelpers";
+import getEstimatedFees from "./js-getFeesForTransaction";
+import estimateMaxSpendable from "./js-estimateMaxSpendable";
+
+const sameFees = (a, b) => (!a || !b ? a === b : a.eq(b));
+
+/**
+ * Create an empty transaction
+ *
+ * @returns {Transaction}
+ */
+export const createTransaction = (): Transaction => ({
+  family: "icon",
+  mode: "send",
+  amount: new BigNumber(0),
+  recipient: "",
+  useAllAmount: false,
+  fees: null,
+});
+
+/**
+ * Prepare transaction before checking status
+ *
+ * @param {IconAccount} a
+ * @param {Transaction} t
+ */
+export const prepareTransaction = async (a: IconAccount, t: Transaction): Promise<Transaction> => {
+  let fees = t.fees;
+
+  fees = await getEstimatedFees({ a, t });
```

Comment:
This doesn't seem necessary?
```suggestion
  const fees = await getEstimatedFees({ a, t });
```